src/chart/heatmap/HeatmapView.ts

```diff
@@ -245,13 +259,19 @@ class HeatmapView extends ChartView {
                     dataDimY
                 ]);
 
+                const shape: Shape = {
+                    x: Math.floor(Math.round(point[0]) - width / 2),
+                    y: Math.floor(Math.round(point[1]) - height / 2),
+                    width: Math.ceil(width),
+                    height: Math.ceil(height)
+                };
+
+                if (borderRadius || Array.isArray(borderRadius)) {
```

Comment:
As described above. The `r` can still be changed after `Rect` created. So it's not necessary to prepare the `shape` with `r` in two different branches. You only need to use

```
rect.shape.r = borderRadius;
```

to update the `r` around the line https://github.com/apache/echarts/blob/04392f5336f0f895c37b3e3f36294b67fc5fdf5c/src/chart/heatmap/HeatmapView.ts#L326

Reply:
Also using `isArray` method in the `zrender/src/core/util` instead of `Array.isArray`.

In fact the `Array.isArray` check can be removed because we usually don't do strict type checking in the logic code. Usually we only do the check that may have exceptions and break the whole app.

For example in this case `borderRadius` we won't reject the value like `{ 0: 5, 1: 5, 2: 5, 3: 5 }` or `TypedArray`.  It make users code can still run even if they made a small mistake. 

Instead we are trying to leave this kind of checking to the TypeScript. Which can provide strict enough type check and won't bring any extra code to the lib.

Reply:
It means, we just need to get the value and assign it, the value about type check uses the TypeScript compiler, bcs we just care about the value. wonderful.